addons/point_of_sale/models/pos_order.py

```diff
@@ -136,7 +136,7 @@ def _link_combo_items(self, combo_child_uuids_by_parent_uuid):
 
     def _process_saved_order(self, draft):
         self.ensure_one()
-        if not draft:
+        if not draft and self.state != 'cancel':
```

Comment:
Is it probably only "draft" order that should pass here, so why not self.state == 'draft' ? Can you investigate